pkg/cluster/fixssh.go

```diff
@@ -35,230 +43,171 @@ func (m *manager) fixSSH(ctx context.Context) error {
 
 	lb, err := m.checkAndUpdateLB(ctx, resourceGroup, lbName)
 	if err != nil {
-		m.log.Warnf("Failed checking and Updating Load Balancer with error: %s", err)
+		m.log.Errorf("Failed checking and Updating Load Balancer with error: %s", err)
 		return err
 	}
 
-	err = m.checkandUpdateNIC(ctx, resourceGroup, infraID, lb)
+	err = m.checkAndUpdateNICsInResourceGroup(ctx, resourceGroup, infraID, &lb)
 	if err != nil {
-		m.log.Warnf("Failed checking and Updating Network Interface with error: %s", err)
 		return err
 	}
 
 	return nil
 }
 
-func (m *manager) checkAndUpdateLB(ctx context.Context, resourceGroup string, lbName string) (lb mgmtnetwork.LoadBalancer, err error) {
-	lb, err = m.loadBalancers.Get(ctx, resourceGroup, lbName, "")
-	if err != nil {
-		return lb, err
-	}
+func (m *manager) checkAndUpdateNICsInResourceGroup(ctx context.Context, resourceGroup string, infraID string, lb *mgmtnetwork.LoadBalancer) (err error) {
+	masterSubnetID := m.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID
+	pagerOpts := &armnetwork.InterfacesClientListOptions{}
+	pager := m.armInterfaces.NewListPager(resourceGroup, pagerOpts)
```

Comment:
We have traditionally constructed wrappers around clients which will fetch all the resources rather than paging through them.  

I don't know if we're doing this today still, but can/should we add an Addon to the interfaces client we have today? 

Something like [this](https://github.com/Azure/ARO-RP/blob/765fc6e6b708d40663fc12f8e203948c46126ea2/pkg/util/azureclient/azuresdk/armnetwork/privatelinkservices_addons.go#L18-L29)

Reply:
10-4.  I'll refactor that bit and add the wrapper as per the example you linked to.